build.js

```diff
@@ -0,0 +1,109 @@
+// This script contains all the steps for generating the output results.
+// This can be triggered by running `npm run build`
+//
+// A majority of this file uses the UglifyJS API
+// https://www.npmjs.com/package/uglify-js
```

Comment:
this new `build.js` replaces the set of scripts that were in the `package.json`. This is a more organized way to bundle and expose all the different scripts that we need, allows us to write comments, and have more consistent behavior across all the generated scripts.

Candidly, this is slightly more error prone, compared to the CLI interface (which is better documented), but all in all, worth the migration.